### PR TITLE
SUBMARINE-278. Add commons-metastore module

### PR DESCRIPTION
**Pull request by LinhaoZhu**
### What is this PR for?
1. Create a new module, This module is used to manage the metadata.
2. Commons-metastore needs to provide metadata services for workbench-server and submarine-server, submarine-sdk.
3. Add 'maven-checkstyle-plugin' and 'maven-enforcer-plugin' to the commons-runtime and commons-utils modules.
4. Modify MockClientContext class that do not conform to code specifications.

### What type of PR is it?
[ Feature ]

### Todos
* [ ] - Task

### What is the Jira issue?
*  [SUBMARINE-278](https://issues.apache.org/jira/browse/SUBMARINE-278)

### How should this be tested?
* [Pass CI](https://travis-ci.org/LinhaoZhu/hadoop-submarine/builds/607559265)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -114,6 +114,7 @@
     <zeppelin.version>0.9.0-SNAPSHOT</zeppelin.version>
     <jgit.version>5.5.1.201910021850-r</jgit.version>
     <atomix.version>3.0.0-rc4</atomix.version>
+    <hive-metastore.version>2.1.1</hive-metastore.version>
```
Comment (liuxunorg) [inline pom.xml]: I think, need change `hive-metastore.version` to `hive.version`.

Comment (LinhaoZhu) [reply pom.xml]: Done.

**Comment by LinhaoZhu** [comment]
> The src license header should be consistent. http://www.apache.org/legal/src-headers.html#headers

**Comment by LinhaoZhu** [comment]
> The src license header should be consistent. http://www.apache.org/legal/src-headers.html#headers

Done.

**Comment by liuxunorg** [comment]
@LinhaoZhu Thank you contributor this feture.
Please provide metadata access services in the following PR. Thank you!

**Pull request closed without merging**